### PR TITLE
[ray] Fix the api called for placement group

### LINKED ISSUE #2064: [Core] Placement group implementation deprecated
The following implementation is deprecated in the ray==2.4.0. We need to fix this.
https://github.com/skypilot-org/skypilot/blob/17a8dc83dbbda2752754e59505c38a13b6a5d739/sky/backends/cloud_vm_ray_backend.py#L402-L403
```
/home/gcpuser/.sky/sky_app/sky_job_1:359: DeprecationWarning: placement_group parameter is deprecated. Use scheduling_strategy=PlacementGroupSchedulingStrategy(...) instead, see the usage at https://docs.ray.io/en/releases-2.4.0/ray-core/package-ref.html
#ray-remote.
  for i in range(pg.bundle_count)
/home/gcpuser/.sky/sky_app/sky_job_1:359: DeprecationWarning: placement_group_bundle_index parameter is deprecated. Use scheduling_strategy=PlacementGroupSchedulingStrategy(...) instead, see the usage at https://docs.ray.io/en/releases-2.4.0/ray-core/pac
kage-ref.html#ray-remote.
  for i in range(pg.bundle_count)
```

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #2064

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch -c min --cloud gcp --cpu 2 echo hi`
  - [x] `sky exec min echo hi`
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [x] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
  - [x] `pytest tests/test_smoke.py::test_job_queue_multinode`
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/backends/cloud_vm_ray_backend.py [hunk 1]
```diff
@@ -359,7 +359,13 @@ def add_gang_scheduling_placement_group_and_setup(
...
                 setup_pg = ray.util.placement_group(setup_bundles, strategy='STRICT_SPREAD')
                 setup_workers = [run_bash_command_with_log \\
-                    .options(name='setup', num_cpus=_SETUP_CPUS, placement_group=setup_pg, placement_group_bundle_index=i) \\
+                    .options(
+                        name='setup',
+                        num_cpus=_SETUP_CPUS,
+                        scheduling_strategy=ray.util.scheduling_strategies.PlacementGroupSchedulingStrategy(
+                            placement_group=setup_pg,
+                            placement_group_bundle_index=i)
+                        ) \\
```
Comment (concretevitamin) [inline sky/backends/cloud_vm_ray_backend.py:368]:
```suggestion
                    ) \\
```

File: sky/backends/cloud_vm_ray_backend.py [hunk 2]
```diff
@@ -449,29 +458,29 @@ def add_ray_task(self,
...
                  f' Found: {ray_resources_dict}.')
             num_gpus = list(ray_resources_dict.values())[0]
-            resources_str = f', resources={json.dumps(ray_resources_dict)}'
+            options.append(f'resources={json.dumps(ray_resources_dict)}')
 
-            # Passing this ensures that the Ray remote task gets
-            # CUDA_VISIBLE_DEVICES set correctly.  If not passed, that flag
-            # would be force-set to empty by Ray.
-            num_gpus_str = f', num_gpus={num_gpus}'
             # `num_gpus` should be empty when the accelerator is not GPU.
```
Comment (concretevitamin) [inline sky/backends/cloud_vm_ray_backend.py:472]: this comment seems out-of-place to me now

**Review by concretevitamin — APPROVED** [review]
Thanks @Michaelvll, LGTM. Since it touches scheduling jobs, maybe we should run the entire smoke tests.

File: sky/backends/cloud_vm_ray_backend.py (hunk 2 above)
Comment (Michaelvll) [reply sky/backends/cloud_vm_ray_backend.py]: Moved. Thanks!

File: sky/backends/cloud_vm_ray_backend.py (hunk 1 above)
Comment (Michaelvll) [reply sky/backends/cloud_vm_ray_backend.py]: Done. Thanks!

**Pull request merged**